client/component.go

```diff
@@ -1,23 +1,107 @@
+// SPDX-License-Identifier: Apache-2.0
+
+// Copyright 2023 PANTHEON.tech
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
 package client
 
-import "go.pantheon.tech/stonework/proto/cnfreg"
+import (
+	"context"
+	"fmt"
+
+	"go.ligato.io/vpp-agent/v3/cmd/agentctl/api/types"
+	"go.ligato.io/vpp-agent/v3/cmd/agentctl/client"
+
+	"go.ligato.io/vpp-agent/v3/proto/ligato/kvscheduler"
+	"go.pantheon.tech/stonework/plugins/cnfreg"
+	cnfregpb "go.pantheon.tech/stonework/proto/cnfreg"
+)
+
+type ComponentMode int32
+
+const (
+	// Foreign means the component is not managed by StoneWork
+	ComponentForeign ComponentMode = iota
```

Comment:
It is good practice to keep the initial value (`0`) for enum like constants to be _unknown_ / _undefined_ / _unset_. This ensures that the default value for the enum will be clearly not yet determined.

Reply:
This is probably very trivial thing, but to me the term "_foreign_" sounds a bit unrelated to other components. Perhaps one of the following might fit better:
- subsidiary
- auxiliary
- accessory

Reply:
Agreed, auxiliary seems better here.